src/main/scala/Silicon.scala

```diff
@@ -242,11 +244,21 @@ class Silicon(val reporter: Reporter, private var debugInfo: Seq[(String, Any)]
 
     val failures =
       results.flatMap(r => r :: r.previous.toList)
-             .collect{ case f: Failure => f } /* Ignore successes */
-             .sortBy(_.message.pos match { /* Order failures according to source position */
-                case pos: ast.HasLineColumn => (pos.line, pos.column)
-                case _ => (-1, -1)
-             })
+        .collect{ case f: Failure => f } /* Ignore successes */
+        .pipe(allResults => {

```

Comment:
I chose pipe here because it allows me to do something entirely different depending on whether the flag for reporting the branch is set. I couldn't find any other way to do "optional function chaining", but if there is a way I'd happily do it that way.